static/js/components/tiles/answer_table_tile.tsx

```diff
@@ -139,49 +163,121 @@ const fetchData = async (
         ? val[0].name || val[0].value || val[0].dcid
         : entity;
     });
+
     // The rest of the promises are for column values
     const propResp = resp.slice(1);
-    const values = {};
+    // map entity -> propExpr -> list of values
+    const values: Record<string, Record<string, string[]>> = {};
+    // map entity -> propExpr -> list of provenanceIds
+    const provIds: Record<string, Record<string, string[]>> = {};
+    // store unique provenanceIds for subsequent API fetch for their URLs
+    const uniqueProvIds = new Set<string>();
+
     props.entities.forEach((entity) => {
       values[entity] = {};
+      provIds[entity] = {};
     });
-    const provIds: Set<string> = new Set();
     propResp.forEach((resp, i) => {
       Object.keys(resp.data).forEach((entity) => {
         const val = resp.data[entity];
-        const entityResults = new Set();
+
+        // Get all values of the current property for the current entity
+        const entityResults = new Set<string>();
         val.forEach((singleVal) => {
           entityResults.add(
             singleVal.name || singleVal.value || singleVal.dcid
           );
-          provIds.add(singleVal.provenanceId);
         });
         values[entity][props.columns[i].propertyExpr] =
           Array.from(entityResults);
+
+        // Get all sources associated with the values
+        const valSources = val.map((singleVal) => {
```

Comment:
nit: call this valProvId instead so that we're consistent about provId vs source

Reply:
Good call out, swapped to valProvId